pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -485,13 +777,23 @@ pub mod pallet {
 				Error::<T>::DomainNotSupported
 			);
 
-			let router =
-				DomainRouters::<T>::get(destination.clone()).ok_or(Error::<T>::RouterNotFound)?;
+			ensure!(
+				DomainRouters::<T>::contains_key(destination.clone()),
+				Error::<T>::RouterNotFound
+			);
 
-			router.send(T::Sender::get(), message.clone())?;
+			<OutboundMessageNonceStore<T>>::try_mutate(|n| {
+				*n = n.ensure_add(T::OutboundMessageNonce::one())?;
+				Ok::<_, DispatchError>(())
```

Comment:
Nit: Could return `*n` here to not have to read the value again below

```suggestion
				Ok::<T:: OutboundMessageNonce, DispatchError>(*n)
```

Reply:
Done.